ecosystem/python/sdk/aptos_sdk/ed25519.py

```diff
@@ -123,6 +161,35 @@ def serialize(self, serializer: Serializer):
         serializer.to_bytes(self.signature)
 
 
+class MultiEd25519Signature:
+    signatures: List[Signature]
+    bitmap: bytes
+
+    def __init__(
+        self,
+        public_key: MultiEd25519PublicKey,
+        signatures_map: List[Tuple[PublicKey, Signature]],
+    ):
+        self.signatures = list()
+        bitmap = 0
+        for entry in signatures_map:
+            self.signatures.append(entry[1])
+            index = public_key.keys.index(entry[0])
+            shift = 31 - index  # 32 bit positions, left to right.
+            bitmap = bitmap | (1 << shift)
+        # 4-byte big endian bitmap.
+        self.bitmap = bitmap.to_bytes(4, "big")
+
+    def to_bytes(self) -> bytes:
+        concatenated_signatures = bytes()
+        for signature in self.signatures:
+            concatenated_signatures += signature.data()
+        return concatenated_signatures + self.bitmap
+
+    def serialize(self, serializer: Serializer):
```

Comment:
would benefit from e2e tests and validation.

Reply:
even if you can't add bytes for tests, we can still sign and verify

Reply:
@davidiw See new tests.